docs/build/micro-rdk/sensor/ultrasonic.md

```diff
@@ -0,0 +1,109 @@
+tags: ["sensor", "components", "ultrasonic"]
+icon: true
+images: ["/icons/components/sensor.svg"]
+# SME: Andrew Morrow
+---
+
+Configure an `ultrasonic` sensor to integrate the [HC-S204](https://www.sparkfun.com/products/15569) ultrasonic distance sensor into your machine:
+
+{{< tabs >}}
+{{% tab name="Config Builder" %}}
+
+Navigate to the **Config** tab of your machine's page in [the Viam app](https://app.viam.com).
+Click on the **Components** subtab and click **Create component**.
+Select the `sensor` type, then select the `ultrasonic` model.
+Enter a name for your sensor and click **Create**.
+
+![Creation of a ultrasonic sensor in the Viam app config builder.](/build/micro-rdk/ultrasonic/ultrasonic-sensor-ui-config.png)
+
+Copy and paste the following attribute template into your sensor's **Attributes** box.
+Then remove and fill in the attributes as applicable to your sensor, according to the table below.
+
+{{< tabs >}}
+{{% tab name="Attributes template" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "trigger_pin": "<pin-number>",
+  "echo_interrupt_pin": "<pin-number>",
+  "timeout_ms": <int>
+}
+```
+
+{{% /tab %}}
+{{% tab name="Attributes example" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "trigger_pin": "15",
+  "echo_interrupt_pin": "18",
+  "timeout_ms": "200"
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{% /tab %}}
+{{% tab name="JSON Template" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "components": [
+    {
+      "name": "<your-ultrasonic-sensor-name>",
+      "model": "ultrasonic",
+      "type": "sensor",
+      "namespace": "rdk",
+      "attributes": {
+        "trigger_pin": "<pin-number>",
+        "echo_interrupt_pin": "<pin-number>",
+        "timeout_ms": <int>
+      },
+      "depends_on": []
+    }
+  ]
+}
+```
+
+{{% /tab %}}
+{{% tab name="JSON Example" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "components": [
+    {
+      "name": "my-ultrasonic-sensor",
+      "model": "ultrasonic",
+      "type": "sensor",
+      "namespace": "rdk",
+      "attributes": {
+        "trigger_pin": "15",
+        "echo_interrupt_pin": "18",
+        "timeout_ms": "200"
+      },
+      "depends_on": []
+    }
+  ]
+}
+```
+
+{{% /tab %}}
+{{% /tabs %}}
+
+The following attributes are available for `ultrasonic` sensors:
+
+<!-- prettier-ignore -->
+| Attribute | Type | Inclusion | Description |
+| --------- | ---- | --------- | ----------- |
+| `trigger_pin` | string | **Required** | The GPIO number of the [board's](/build/micro-rdk/board/) GPIO pin that you have wired to the trigger pin of your ultrasonic sensor. |
+| `echo_interrupt_pin` | string | **Required** | The GPIO number of the board's GPIO pin that you have wired to the echo pin of your ultrasonic sensor. |
+| `timeout_ms`  | int | Optional | Time to wait in milliseconds before initiating a timeout when requesting readings from your ultrasonic sensor. <br> Default: `1000`. |
```

Comment:
The default timeout for the micro-rdk usonic sensor is actually 50ms, and it is clamped to 100ms.

Reply:
What do you mean by clamped, it can't go any higher than?

Reply:
That's right. Even if you put in 1000 it will internally limit the timeout to 100ms. The implementation in the micro-rdk is currently blocking. The sensor device itself never needs more than, say, 60ms. But if you set the timeout to 10000ms or something you would basically hang the micro-rdk, so we ignore any timeout above 100ms.